From c63e6c87ff25a263fd7f2fec2f523b1093f4ff67 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 5 May 2015 10:32:04 +0100
Subject: [PATCH 3/4] link ebtables with --no-as-needed and adjust the link
 order to fix runtime crash
Organization: Intel Corporation (UK) Ltd. - Co. Reg. #1134945 - Pipers Way, Swindon SN3 1RJ

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6ad7325..50c1a2b 100644
--- a/Makefile
+++ b/Makefile
@@ -90,7 +90,7 @@ libebtc.so: $(OBJECTS2)
 	$(CC) -shared $(LDFLAGS) -Wl,-soname,libebtc.so -o libebtc.so -lc $(OBJECTS2)
 
 ebtables: $(OBJECTS) ebtables-standalone.o libebtc.so
-	$(CC) $(CFLAGS) $(CFLAGS_SH_LIB) $(LDFLAGS) -o $@ ebtables-standalone.o -I$(KERNEL_INCLUDES) -L. -Lextensions -lebtc $(EXT_LIBSI) \
+	$(CC) $(CFLAGS) $(CFLAGS_SH_LIB) $(LDFLAGS) -o $@ ebtables-standalone.o -I$(KERNEL_INCLUDES) -L. -Lextensions -Wl,--no-as-needed $(EXT_LIBSI) -lebtc \
 	-Wl,-rpath,$(LIBDIR)
 
 ebtablesu: ebtablesu.c
-- 
2.1.0

